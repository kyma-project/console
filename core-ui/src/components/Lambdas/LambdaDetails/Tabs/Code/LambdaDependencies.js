import React from 'react';
import PropTypes from 'prop-types';
import { ControlledEditor } from '@monaco-editor/react';
import { CollapsiblePanel } from 'react-shared';

const LambdaDependencies = ({ dependencies, setDependencies }) => {
  return (
    <CollapsiblePanel
      title="Dependencies"
      children={
        <ControlledEditor
          id="lambdaDependencies"
          height="10em"
          language="json"
          theme="vs-light"
          value={dependencies}
          onChange={(_, value) => setDependencies(value)}
        />
      }
      isOpenInitially={false}
      className="fd-has-margin-medium"
    />
  );
};

LambdaDependencies.propTypes = {
  dependencies: PropTypes.string.isRequired,
  setDependencies: PropTypes.func.isRequired,
};

export default LambdaDependencies;
