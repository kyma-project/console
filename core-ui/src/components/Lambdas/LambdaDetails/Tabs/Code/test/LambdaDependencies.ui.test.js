import React from 'react';
import renderer from 'react-test-renderer';

import LambdaDependencies from '../LambdaDependencies';

describe('Lambda Dependencies', () => {
  it('Render with minimal props', () => {
    const component = renderer.create(
      <LambdaDependencies
        dependencies="dependencies"
        setDependencies={() => {}}
      />,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
