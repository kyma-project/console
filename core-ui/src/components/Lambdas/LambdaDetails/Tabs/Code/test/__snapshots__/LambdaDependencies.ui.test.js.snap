// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lambda Dependencies Render with minimal props 1`] = `
<div
  className="fd-panel collapsible-panel fd-has-margin-medium"
>
  <div
    className="fd-panel__header"
  >
    <div
      className="fd-panel__head"
    >
      <h3
        className="fd-panel__title"
      >
        Dependencies
      </h3>
    </div>
    <div
      className="fd-panel__actions"
    >
      <button
        className="fd-button--light sap-icon--navigation-down-arrow"
        onClick={[Function]}
      />
    </div>
  </div>
  <div
    className="fd-panel__body body body--closed"
  >
    <section
      style={
        Object {
          "display": "flex",
          "height": "10em",
          "position": "relative",
          "textAlign": "initial",
          "width": "100%",
        }
      }
    >
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "height": "100%",
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        Loading...
      </div>
      <div
        style={
          Object {
            "display": "none",
            "width": "100%",
          }
        }
      />
    </section>
  </div>
</div>
`;
